### PR TITLE
Slot machine

### PR DESCRIPTION
**Pull request by PikalaxALT**

### CODE REVIEW

**Comment by PikalaxALT** [comment]
2x NONMATCHING

File: include/menu.h [hunk 1]
```diff
@@ -24,6 +24,9 @@ void AddTextPrinterParameterized5(u8 windowId, u8 fontId, const u8 *str, u8 x, u
 void sub_812E6DC(u8 windowId, const u8 * src, u16 x, u16 y);
 void StartBlendTask(u8 eva_start, u8 evb_start, u8 eva_end, u8 evb_end, u8 ev_step, u8 priority);
 bool8 IsBlendTaskActive(void);
+void AddItemMenuActionTextPrinters(u8 windowId, u8 fontId, u8 left, u8 top, u8 letterSpacing, u8 lineHeight, u8 itemCount, const struct MenuAction *strs, const u8 *a8);
```
Comment (jiangzhengwenjz) [inline include/menu.h]: It's already at line 46

File: include/menu.h [hunk 2]
```diff
@@ -24,6 +24,9 @@ void AddTextPrinterParameterized5(u8 windowId, u8 fontId, const u8 *str, u8 x, u
...
+void sub_810F260(u8 windowId, u8 a1);
```
Comment (jiangzhengwenjz) [inline include/menu.h]: ClearDialogWindowAndFrameToTransparent, is also declared below

File: include/menu.h [hunk 3]
```diff
@@ -24,6 +24,9 @@ void AddTextPrinterParameterized5(u8 windowId, u8 fontId, const u8 *str, u8 x, u
...
+void sub_812E62C(u8 windowId, u8 fontId, const u8 *str, u8 x, u8 y, u8 speed, void (*callback)(struct TextPrinterTemplate *, u16), u8 letterSpacing, u8 lineSpacing);
```
Comment (jiangzhengwenjz) [inline include/menu.h]: AddTextPrinterParameterized5, is declared above

File: src/slot_machine.c [hunk 4]
```diff
@@ -0,0 +1,2886 @@
...
+    u8 field_205C[0x800];
+}; // size: 285C
+
+struct UnkStruct_8466C0C
+{
+    const u16 * tiles;
+    u32 count;
+};
+
+EWRAM_DATA struct SlotMachineState * sSlotMachineState = NULL;
```
Comment (jiangzhengwenjz) [inline src/slot_machine.c]: many variables and functions should be static

**Pull request merged**